File: Design_Patterns/Observer_Design_pattern/java/Observer/TVDisplay.java
package Observer;

import Observable.WeatherStation;

/**
 * TVDisplay - Concrete Observer Implementation
 * 
 * This class represents a TV that displays weather information.
 * It observes the WeatherStation and updates its display when notified.
 * 
 * Key Features:
 * - Implements Observer interface
 * - Maintains reference to the WeatherStation (Observable)
 * - Updates display when notified of temperature changes
 */
public class TVDisplay implements Observer {
    private WeatherStation weatherStation;

    public TVDisplay(WeatherStation weatherStation) {
        this.weatherStation = weatherStation;
    }

    @Override
    public void update() {
        int currentTemp = weatherStation.getTemperature();
        display(currentTemp);
    }

    private void display(int temperature) {
        System.out.println("📺 TV Display: Weather Update - Temperature: " +
                temperature + "°C");
    }
}

File: Design_Patterns/BehaviouralDesignPattern/Observer_Design_pattern/cpp/TVDisplay.cpp
/**
 * TVDisplay.cpp
 * 
 * Implementation of TVDisplay class
 */

#include "TVDisplay.h"
#include <iostream>

TVDisplay::TVDisplay(WeatherStation* ws) : weatherStation(ws) {}

void TVDisplay::update() {
    int currentTemp = weatherStation->getTemperature();
    display(currentTemp);
}

void TVDisplay::display(int temperature) {
    std::cout << "📺 TV Display: Weather Update - Temperature: " 
              << temperature << "°C" << std::endl;
}
